lib/xctest.js

```diff
@@ -0,0 +1,185 @@
+import { startHouseArrestService, startTestmanagerdService, startInstallationProxyService, startInstrumentService } from './services';
+import { INSTRUMENT_CHANNEL } from './instrument';
+import { TESTMANAGERD_CHANNEL } from './testmanagerd';
+import { DTXMessageAuxBuffer } from './instrument/headers';
+import { NSUUID, XCTestConfiguration } from './instrument/transformer/nskeyed';
+import { uuidV4 } from '@appium/support/build/lib/util';
+import { getOSVersion } from './utilities';
+import log from './logger';
+import B from 'bluebird';
+
+class Xctest {
+    constructor(udid, xctestBundleId, targetBundleId = undefined, productModuleName = 'WebDriverAgentRunner') {
+        this.udid = udid;
+        this.running = false;
+        this._executing = false;
+        this.xctestBundleId = xctestBundleId;
+        this.targetBundleId = targetBundleId;
+        this.productModuleName = productModuleName;
+    }
+
+    async _launchAppRunner(majorVersion, sessionIdentifier) {
+        const { PROCESS_CONTROL } = INSTRUMENT_CHANNEL;
+        const installation = await startInstallationProxyService(this.udid);
+        let lookupResult;
+        try {
+            await installation.lookupApplications({ bundleIds: [this.xctestBundleId] });
+        } finally {
+            installation.close();
+        }
+        const appInfo = lookupResult[this.xctestBundleId];
+        if (!appInfo) {
+            log.error(`${this.xctestBundleId} not found!`);
+            return;
+        }
+        const signerIdentifier = appInfo.SignerIdentity;
+        log.debug('SignerIdentifier:', signerIdentifier);
+        const appContainer = appInfo.Container;
+        const execName = appInfo.CFBundleExecutable;
+        if (!execName.endsWith('-Runner')) {
+            log.error('Invalid CFBundleExecutable:', execName);
+            return;
+        }
+        const targetName = execName.substr(0, execName.indexOf('-Runner'));
+        const xctestPath = `/tmp/${targetName}-${sessionIdentifier.toUpperCase()}.xctestconfiguration`;
```

Comment:
please use  `tempDir.openDir()` helper from `@appium/support` to create a temporary folder and then remove it when it is not needed anymore